src/backend/distributed/commands/dependencies.c

```diff
@@ -55,29 +100,51 @@ static char * DropTableIfExistsCommand(Oid relationId);
  * postgres native CREATE IF NOT EXISTS, or citus helper functions.
  */
 static void
-EnsureDependenciesExistOnAllNodes(const ObjectAddress *target)
+EnsureRequiredObjectExistOnAllNodes(const ObjectAddress *target,
+									RequiredObjectSet requiredObjectSet)
 {
-	List *dependenciesWithCommands = NIL;
+	Assert(requiredObjectSet == REQUIRE_ONLY_DEPENDENCIES ||
+		   requiredObjectSet == REQUIRE_OBJECT_AND_DEPENDENCIES);
+
```

Comment:
Before doing anything else, here we can first check if the target object is in fact distributed: if it is  we can return early here, regardless of the `RequiredObjectSet`

Reply:
I tested to put IsAnyObjectDistributed method however got test failures so I removed

Reply:
I believe lastly we agreed on adding the following check?

```suggestion
	if (IsAnyObjectDistributed(list_make1((ObjectAddress *) target)))
	{
		return;
	}
```

Reply:
This is the test error when we use IsAnyObjectDistributed(list_make1((ObjectAddress *) target))